### PR TITLE
Decouple "push" and "down" implementations

### LINKED ISSUE #777: "okteto push" does not work if "okteto up" is active
**Describe the bug**
The issue only happens for clusters not managed by okteto.

### PR DESCRIPTION
**Pull request by pchico83**
Fixes #777

### CODE REVIEW

File: pkg/k8s/deployments/crud.go [hunk 1]
```diff
@@ -204,30 +204,21 @@ func HasBeenChanged(d *appsv1.Deployment) bool {
...
-	}
-	if imageToRedeploy != "" {
-		for _, rule := range tr.Rules {
-			devContainer := GetDevContainer(&d.Spec.Template.Spec, rule.Container)
-			if devContainer == nil {
-				return fmt.Errorf("Container '%s' not found in deployment '%s'", rule.Container, d.GetName())
-			}
-
-			devContainer.Image = imageToRedeploy
+// UpdateDeployments update all deployments in the given translaation list
```
Comment (rberrelleza) [inline pkg/k8s/deployments/crud.go]:
```suggestion
// UpdateDeployments update all deployments in the given translation list
```

File: pkg/k8s/deployments/crud.go [hunk 2]
```diff
@@ -204,30 +204,21 @@ func HasBeenChanged(d *appsv1.Deployment) bool {
...
+func UpdateDeployments(trList map[string]*model.Translation, c *kubernetes.Clientset) error {
+	for _, tr := range trList {
+		if tr.Deployment == nil {
```
Comment (rberrelleza) [inline pkg/k8s/deployments/crud.go]: why would deployment be nil?

File: pkg/k8s/deployments/crud.go [hunk 3]
```diff
@@ -204,30 +204,21 @@ func HasBeenChanged(d *appsv1.Deployment) bool {
...
 		}
-	}
-	if err := update(d, c); err != nil {
-		return err
 	}
 	return nil
 }
 
-func translateDevModeOff(d *appsv1.Deployment) (*appsv1.Deployment, error) {
+//TranslateDevModeOff reverse the dev mode translation
```
Comment (rberrelleza) [inline pkg/k8s/deployments/crud.go]:
```suggestion
//TranslateDevModeOff reverses the dev mode translation
```

File: pkg/k8s/deployments/crud.go (hunk 2 above)
Comment (pchico83) [reply pkg/k8s/deployments/crud.go]: It is for the case it is created by `up` or `push`

**Pull request merged**